src/manifest.c

```diff
@@ -538,12 +559,30 @@ static int retrieve_manifests(int current, int version, char *component, struct
 	free(dir);
 
 	/* FILE is not set for a MoM, only for bundle manifests */
-	if (file && current < version) {
+	if (file && current < version && strcmp(component, "full") != 0) {
 		if (try_delta_manifest_download(current, version, component, file) == 0) {
 			return 0;
 		}
 	}
 
+	/* If it's mix content just hardlink instead of curl download */
+	if (is_mix) {
+		string_or_die(&filename, "%s/%i/Manifest.%s.tar", state_dir, version, component);
+		string_or_die(&url, "%s/%i/Manifest.%s.tar", basedir, version, component);
+		ret = link(url, filename);
+		/* Try doing a regular rename if hardlink fails */
+		if (ret) {
+			ret = rename(url, filename);
```

Comment:
There is no error handling for a `rename()` failure, so I think that should at least be addressed. If errors are handled by fallthrough, I would add a code comment.

Reply:
Yes they are handled by fallthrough and there's a comment on https://github.com/clearlinux/swupd-client/blob/mixer-integration/src/manifest.c#L584 about it, but that may not be the best place for it. I added another in the earlier location as well to be more clear.

Reply:
Ah, okay, I see that now. I didn't look too closely at the surrounding code block for context. Maybe the comment on line 568 could be expanded so that reading the function top to bottom will be more clear?